Add ReadTagApp

HugoOliveiraSoares opened the pull request:
Hello!

This pull request implements a new feature that allows reading information from an NFC tag using the PN532 module as a reader.
With this feature it is possible to read: 
- UID
- ATQ
- SAK

Uses Seeed-Studio's [PN532](https://github.com/Seeed-Studio/PN532) libraries and [NDEF](https://github.com/don/NDEF) by Don Coleman that were added to the lib directory

HugoOliveiraSoares committed:
add PN532 and NDEF libs

HugoOliveiraSoares committed:
read tag uid

HugoOliveiraSoares committed:
change variables names for i2c

HugoOliveiraSoares committed:
Definition of SDA and SCL pins moved to platform.ini file

marivaaldo commented on src/applications/nfc/ReadTag/ReadTagRun.h:
Please rename to `ReadTagRunScreen.h`

marivaaldo commented on src/applications/nfc/ReadTag/ReadTagRun.cpp:
Please rename to `ReadTagRunScreen.cpp`

marivaaldo commented on src/domain/entities/DeviceBase.cpp:
Please add `i2cInterface` to `begin` method.

marivaaldo commented on src/domain/entities/I2CInterfaceBase.h:
You can use `I2CInterfaceBase` (using `DSDA_PIN` and `DSCL_PIN`) like `SDCardInterfaceBase` to not needs implement I2CInterface on each device.

marivaaldo commented on platformio.ini:
I'ts possible to install libs `NDEF, PN532, etc` in this file?

marivaaldo reviewed (changes requested):
Please see comments. Thank you so much!

HugoOliveiraSoares committed:
move PN532 and NDEF libs to platformio.ini file

HugoOliveiraSoares committed:
Rename ReadTagRun to ReadTagRunScreen

renamed:    src/applications/nfc/ReadTag/ReadTagRun.cpp -> src/applications/nfc/ReadTag/ReadTagRunScreen.cpp
renamed:    src/applications/nfc/ReadTag/ReadTagRun.h -> src/applications/nfc/ReadTag/ReadTagRunScreen.h

HugoOliveiraSoares committed:
add i2cInterface to begin method

HugoOliveiraSoares committed:
removes unnecessary I2CInterface implementation for each device

HugoOliveiraSoares commented:
Hi @marivaaldo, the problems found have been fixed. Thanks for the review.

marivaaldo reviewed (changes requested):
Please, you can rebase with `develop` and change PR to merge into `develop` branch? Sorry to not comment this previously and thanks for you great job!

HugoOliveiraSoares commented:
> Please, you can rebase with `develop` and change PR to merge into `develop` branch? Sorry to not comment this previously and thanks for you great job!

Sure, I will rebase the branch with `develop`

HugoOliveiraSoares committed:
adapt ReadTagApp to system theming
